servers/src/main/java/tachyon/worker/file/FileDataManager.java

```diff
@@ -75,13 +77,20 @@ public synchronized void persistFile(long fileId, List<Long> blockIds) throws IO
     OutputStream outputStream = mUfs.create(dstPath);
     final WritableByteChannel outputChannel = Channels.newChannel(outputStream);
 
+    Map<Long, Long> blockIdToLockId = Maps.newHashMap();
+    // lock all the blocks to prevent any eviction
     for (long blockId : blockIds) {
       long lockId;
       try {
         lockId = mBlockDataManager.lockBlock(Sessions.CHECKPOINT_SESSION_ID, blockId);
+        blockIdToLockId.put(blockId, lockId);
       } catch (BlockDoesNotExistException e) {
         throw new IOException(e);
       }
+    }
```

Comment:
also, we need to be very cautious here to avoid dead-lock: is it possible for the worker to have some other thread is also requesting to lock the same set of blocks, but with a different order? In that case, we may endup with deadlock.